tests/test_askrene.py

```diff
@@ -451,6 +442,43 @@ def test_fees_dont_exceed_constraints(node_factory):
     assert amount <= max_msat
 
 
+def test_sourcefree_on_mods(node_factory, bitcoind):
```

Comment:
I am not sure what is being tested here. A single hop payment does not pay fees, why does it matter if `auto.sourcefree` is used or not?

Reply:
A single hop payment *will* pay fees unless you apply auto.sourcefree.  The only logic removing local fees is `auto.sourcefree` overriding the channels (technically, this is incorrect, if we had a loop, but we don't).

This simplifies the path logic, and allows users to ask for partial routes, too.